tests/unit/dataquery/test_api.py

```diff
@@ -1,33 +1,222 @@
+        # time-series. To determine if a time-series is valid, pass through each date and
+        # confirm that the associated value is not a NoneType. If all dates contain NaN
+        # values, exclude the ticker from the DataFrame. For instance, USD_FXXR_NSA would
+        # be removed.
+
+        # All tickers held in the dictionary are valid tickers. Therefore, confirm the
+        # keys for the two dictionary, received & returned, match.
+        results_dict = api.Interface.valid_ticker(_dict=self.results_dict,
+                                                  suppress_warning=True,
+                                                  debug=False)
+        self.assertTrue(len(results_dict.keys()) == len(self.results_dict.keys()))
+
+        test = sorted(list(results_dict.keys()))
+        benchmark = sorted(list(self.results_dict.keys()))
+        self.assertTrue(test == benchmark)
+
+        # Add a ticker that does not have a "valid" time-series and will subsequently be
+        # removed. Confirm the series has been removed from the dictionary.
+        f_ticker = next(iter(results_dict.keys()))
+        shape = results_dict[f_ticker].shape
+        # Again, as described above, a series is not valid if all values are NoneType.
+        data = np.array([None] * (shape[0] * shape[1]))
+
+        results_dict['DB(JPMAQS,USD_FXXR_NSA'] = data.reshape(shape)
+        results_dict_USD = api.Interface.valid_ticker(self.results_dict,
+                                                      suppress_warning=True,
+                                                      debug=False)
+        # Ticker should be removed from the dictionary.
+        self.assertTrue('DB(JPMAQS,USD_FXXR_NSA' not in results_dict_USD.keys())
+
+    def test_dataframe_wrapper(self):
+
+        cids_dmca = ['AUD', 'CAD', 'CHF', 'EUR', 'GBP', 'JPY']
+        xcats = ['EQXR_NSA', 'FXXR_NSA']
+
+        tickers = [cid + '_' + xcat for xcat in xcats for cid in cids_dmca]
+
+        # After the time-series have been isolated for each ticker and all the tickers
+        # have been validated, aggregate the results, still held in a dictionary, into
+        # a single DataFrame where the columns will be the passed metrics, 'value' &
+        # 'grading' plus the standardised JPMaQS columns: 'cid', 'xcat', 'real_date'.
+
+        # The method api.Interface.valid_ticker() is not required given each ticker will
+        # be valid by design.
+        self.test_isolate_timeseries()
+
+        results_dict = self.results_dict
+        trial_df = api.Interface.dataframe_wrapper(_dict=results_dict, no_metrics=2,
+                                                   original_metrics=['value', 'grading'])
+
+        # Confirm the dictionary is a standardised DataFrame plus the respective metrics
+        # passed.
+        expected_columns = ['cid', 'xcat', 'real_date', 'value', 'grading']
+        self.assertTrue(sorted(list(trial_df.columns)) == sorted(expected_columns))
```

Comment:
Re-writing the check to `self.assertEqual` that way if it fails we will be told what the difference is?

```python
self.assertEqual(sorted(expected_columns), sorted(list(trial_df.columns)))
```
Also for some reason convention for pytest/unittest seems to be that the "truth" comes first, and then the test value second.